packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java

```diff
@@ -517,6 +520,36 @@ public void onResponse(Call call, Response response) throws IOException {
     }
   }
 
+  @ExpoMethod
+  public void getContentUriAsync(String uri, Promise promise){
+    try{
+      final Uri fileUri = Uri.parse(uri);
+      ensurePermission(fileUri, Permission.WRITE);
+      ensurePermission(fileUri, Permission.READ);
+      checkIfFileDirExists(fileUri);
+      if("file".equals(fileUri.getScheme())){
+        File file = uriToFile(fileUri);
+        Bundle result = new Bundle();
+        result.putString("uri", cUriFromFile(file).toString());
+        promise.resolve(result);
+      }
+
+    }
+    catch (Exception e) {
+      Log.e(TAG, e.getMessage());
+      promise.reject(e);
+    }
+  }
+
+  private Uri cUriFromFile(File file) {
+    try {
+      Application application = mModuleRegistry.getModule(ActivityProvider.class).getCurrentActivity().getApplication();
+      return FileProvider.getUriForFile(application, application.getPackageName() + ".provider", file);
+    } catch (Exception e) {
+      return Uri.fromFile(file);
```

Comment:
i think in this case we actually want to pass the exception back to the user, rather than falling back to returning a `file://` uri.

Reply:
That's more logic!